contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -559,6 +567,20 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
     emit DONAdded(id, isPublic);
   }
 
+  /// @notice Removes DONs from the Capability Registry
+  /// @param donIds The IDs of the DON to be removed
+  function removeDONs(uint32[] calldata donIds) external onlyOwner {
+    for (uint256 i; i < donIds.length; ++i) {
+      uint32 donId = donIds[i];
+      DON storage don = s_dons[donId];
+
+      // DON config count starts at index 1
+      if (don.configCount == 0) revert DONDoesNotExist(donId);
+      delete s_dons[donId];
```

Comment:
I think we can get away with not cleaning up the nested mappings.  Resetting `configCount` to `0` will return empty values in the getter functions. [Test](https://github.com/smartcontractkit/chainlink/pull/13220/files#diff-8e2a85ca0ab438d8736619c7a0eab4c800ef49e1ac8e512e9bd4ac9e9a60e933R80).